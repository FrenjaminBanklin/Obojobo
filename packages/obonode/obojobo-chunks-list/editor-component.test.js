import React from 'react'
import renderer from 'react-test-renderer'

import List from './editor-component'

jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/with-slate-wrapper',
	() => item => item
)
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/editor-component',
	() => props => <div>{props.children}</div>
)
jest.mock('slate-react', () => ({ ReactEditor: { findPath: jest.fn() } }))
jest.mock('slate', () => ({
	Editor: {
		withoutNormalizing: (editor, cb) => {
			cb()
		},
		nodes: jest.fn()
	},
	Transforms: {
		setNodes: jest.fn()
	}
}))

describe('List Editor Node', () => {
	test('List component when not selected', () => {
		const props = {
			selected: false,
			editor: {},
			element: {
				content: {
					listStyles: {}
				}
			}
		}
		const component = renderer.create(<List {...props} />)
		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
