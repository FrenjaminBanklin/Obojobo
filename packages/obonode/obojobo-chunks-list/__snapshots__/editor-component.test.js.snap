// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Editor Node List component when not selected 1`] = `
<div>
  <div
    className="text-chunk obojobo-draft--chunks--list pad"
  />
</div>
`;
